sampuru-server/src/main/java/ca/on/oicr/gsi/sampuru/server/DBConnector.java

```diff
@@ -100,92 +102,106 @@ public void writeDeliverables(JSONArray deliverableArray, String username) throw
-                                checkWritePermission(nextDeliverable.get("location"), username),
-                                checkWritePermission(nextDeliverable.get("notes"), username),
-                                checkWritePermission(PostgresDSL.localDateTime(nextDeliverable.get("expiry_date").toString()), username)
-                        );
-
+                        Row4<String, String, String, LocalDateTime> row = DSL.row(nextDeliverable.get("project_id").toString(), nextDeliverable.get("location").toString(), nextDeliverable.get("notes").toString(), LocalDateTime.parse(nextDeliverable.get("expiry_date").toString(), ServiceUtils.DATE_TIME_FORMATTER));
+                        if(writePermission) {
+                            formattedUnknownDeliverables.add(row);
+                        }
                     }
-                    Result<Record> ids = deliverableInsertValuesStepN.returningResult(DELIVERABLE_FILE.ID).fetch();
 
-                    // Associate the new ids with the appropriate case ids and write to deliverables_case table
-                    for(int i = 0; i < ids.size(); i++){
-                        Integer thisId = Integer.valueOf(ids.get(i).get("id").toString());
+
+                    // Jooq 3.15 introduced .valuesOfRows for adding collection of values to Insert statements safely
+                    Result<Record1<Integer>> deliverableIds = PostgresDSL.using(configuration).insertInto(DELIVERABLE_FILE, DELIVERABLE_FILE.PROJECT_ID, DELIVERABLE_FILE.LOCATION, DELIVERABLE_FILE.NOTES, DELIVERABLE_FILE.EXPIRY_DATE)
+                        .valuesOfRows(formattedUnknownDeliverables)
+                        .returningResult(DELIVERABLE_FILE.ID).fetch();
+
+                    // Associate the new ids with the appropriate case ids and write to deliverable_case table
+                    List<Row2<String, Integer>> deliverableCase = new ArrayList<>();
+                    for(int i = 0; i < deliverableIds.size(); i++) {
+                        Integer deliverableId = Integer.valueOf(Objects.requireNonNull(deliverableIds.get(i).get("id")).toString());
                         JSONArray casesForId = ((JSONArray) ((JSONObject)unknownDeliverables.get(i)).get("case_id"));
-                        for(Object obj: casesForId){
-                            String strObject = (String) obj;
-                            deliverableCaseInsertValuesStepN = deliverableCaseInsertSetStep.values(
-                                    checkWritePermission(thisId, username),
-                                    checkWritePermission(strObject, username)
-                            );
+                        for(Object obj: casesForId) {
+                            String caseId = (String) obj;
+                            Row2<String, Integer> row = DSL.row(caseId, deliverableId);
+                            if(writePermission) {
```

Comment:
Rather than checking this bool that will have the same value every iteration of the for loop on line 132, could this check wrap the for loop?